internal/server/models/provideruser.go

```diff
@@ -8,10 +8,8 @@ import (
 
 // ProviderUser is a cache of the provider's user and their groups, plus any authentication-specific information for that provider.
 type ProviderUser struct {
-	Model
```

Comment:
This may cause problems when deleting provider users since gorm updates `deleted_at` column rather than deleting the row. It's possible the gorm will hard delete the row if the `deleted_at` is not found but that means we will not be able to recover deleted rows. I'm not sure that's a good idea at this time.

Reply:
That's actually the intended behavior in this case. Gorm has some limitations around setting the unique index that make soft-delete here a pain. We can't set the "deleted at" to be part of the primary key since it is nullable and on a separate struct. Then if we try to set this manually, we hit issues with Gorm automatically resetting the primary key on migration.